code/_core/obj/item/tempering/tempering_quality.dm

```diff
@@ -16,38 +16,43 @@
 
 /obj/item/tempering/quality/can_temper(var/mob/activator,var/obj/item/I)
 
-	if(I.quality <= -1)
+	if(!I.can_adjust_quality())
 		activator.to_chat(span("warning","\The [I.name] cannot be improved!"))
 		return FALSE
 
-	if(I.quality >= maximum)
-		activator.to_chat(span("warning","\The [I.name] cannot be improved any further!"))
-		return FALSE
-
 	return ..()
 
 /obj/item/tempering/quality/get_examine_list(var/mob/examiner)
 	. = ..()
-	. += span("notice","Increases the quality of [item_type_name] by [increase]%, up to [maximum]%, with a minimum of [minimum]%.")
+	. += span("notice","Increases the quality of [item_type_name] by [increase]%, with a minimum of [minimum]%.")
 
-/obj/item/tempering/quality/on_temper(var/mob/activator,var/obj/item/I)
-	I.quality = clamp(I.quality + increase,minimum,maximum)
-	. = ..()
+/obj/item/tempering/quality/on_temper(var/mob/activator, var/obj/item/I)
+	I.adjust_quality((I.quality + increase) > minimum ? increase : minimum - I.quality)
```

Comment:
I feel that this can be more readable with

```
if(I.quality + increase < minimum)
 I.adjust_quality(I.quality - minimum) //Add missing quality to force it to minimum.
else
 I.adjust_quality(increase) //Directly add quality.
```

Reply:
fair